Fixes #1791: Handling with inline img

rjayroso opened the pull request:
## Issue This PR Addresses
Fixes #1791
Fixes #1412

## Type of Change
- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description
I piggy-backed off of @huynguyez 's solution that he posted in issue #1791 ([see here](https://github.com/Seneca-CDOT/telescope/issues/1791#issuecomment-784429056)) and just added an extra line `vertical-align:50%;` to center the inline images.

| Before |
| :---: | 
| ![Capture4](https://user-images.githubusercontent.com/48869737/110953539-7aa33400-8315-11eb-85ca-e2329e484127.PNG) |

| After | 
| :---: |
| ![Capture3](https://user-images.githubusercontent.com/48869737/110953642-9c9cb680-8315-11eb-9a46-9430ff14805b.PNG) |

## Checklist
- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

izhuravlev commented:
I was not sure how to test this, so I opened the deployment and saw this:
![image](https://user-images.githubusercontent.com/36719387/110955018-31c19e80-835a-11eb-925d-7bf6aa182fda.png)

huynguyez commented:
> I was not sure how to test this, so I opened the deployment and saw this:
> ![image](https://user-images.githubusercontent.com/36719387/110955018-31c19e80-835a-11eb-925d-7bf6aa182fda.png)
> 
> So I assume some of the regular blogpost images are getting shrunk as well, which is not great.

@izhuravlev Yes I mentioned this issue on Tuesday about this

huynguyez reviewed (changes requested):
Like I talked about this in the last call.

Using ```telescope-post-content div``` will also affect those blogs from blogspot.

izhuravlev commented:
> Like I talked about this in the last call.
> 
> Using `telescope-post-content div` will also affect those blogs from blogspot.

Oh that's what you meant, I just was not sure what blogspot is

rjayroso commented:
> I was not sure how to test this, so I opened the deployment and saw this:
> ![image](https://user-images.githubusercontent.com/36719387/110955018-31c19e80-835a-11eb-925d-7bf6aa182fda.png)
> 
> So I assume some of the regular blogpost images are getting shrunk as well, which is not great.

Hmm, do you think we should just keep what we have now and scrap this along with closing the addressed issue?

izhuravlev commented:
@rjayroso I think we should leave this for now, until any other idea pops up

yuanLeeMidori commented:
> @rjayroso I think we should leave this for now, until any other idea pops up

Agree. I believe currently there is no one method to take care of all the cases.

rjayroso commented:
I removed the styling for the images to be inline as it would, unfortunately, ruin some posts by making the images too small. I'm thinking of keeping it to what we had before but implement the styling for the GitHub links so it matches the font size as the rest of the post (1.8rem). This would at least make the links look less out of place, but it doesn't really deal with the images being too big (see below):

## Before (i.e. No div > h1 styling)
![image](https://user-images.githubusercontent.com/48869737/111284069-27cbc400-8616-11eb-8959-2b95eb7e07e0.png)

## After (set to 1.8rem)
![image](https://user-images.githubusercontent.com/48869737/111283972-0c60b900-8616-11eb-9c9b-c14bb07c939b.png)

birtony reviewed (changes requested):
I think this image on the bottom of @yuanLeeMidori's post does not display properly:
![image](https://user-images.githubusercontent.com/33902374/111322668-89ecef00-863f-11eb-8d99-91c3d996fa86.png)
![image](https://user-images.githubusercontent.com/33902374/111322788-a852ea80-863f-11eb-9b95-602070dd3302.png)
I think we should address it in this issue @rjayroso

rjayroso commented:
@birtony I checked it out and I agree the photo was not being displayed properly. I rebased and added a max-height to make tall pictures more friendly to view, see below (ignore red border):

| **a. Before** | 
| :---: |
| ![beforepic](https://user-images.githubusercontent.com/48869737/111460567-823a5280-86f2-11eb-8def-6ee95eac9ae2.PNG) |
| **a. After** |
| ![afterpic](https://user-images.githubusercontent.com/48869737/111460597-8bc3ba80-86f2-11eb-9ecd-a1c28621a39f.PNG) |
| **b. Before** |
| ![beforepic2](https://user-images.githubusercontent.com/48869737/111460690-a433d500-86f2-11eb-98cc-9ce1a83d17a0.PNG) |
| **b. After** | 
| ![afterpic2](https://user-images.githubusercontent.com/48869737/111460709-aac24c80-86f2-11eb-8dc6-c5e1bf76254d.PNG) |
  
I think the photo on the bottom of Yuan's post is a bit trippy because of how it was screenshotted, but either way let me know if you like this change otherwise I can revert it.

HyperTHD commented:
Make sure you rebase this before you merge this, looks great

birtony commented on src/web/src/styles/telescope-post-content.css:
```diff
@@ -16,11 +16,33 @@
   height: auto;
 }
+
+.telescope-post-content div > a > img {
+  width: 32px;
+  display: inline;
+  vertical-align: middle;
+}
+
+.telescope-post-content h1 > a > img {
```
You could group these like:
```css
.telescope-post-content div, h1 > a > img {
...
}
```

rjayroso replied:
I tried this just now and it seems to ignore the CSS completely (as if it ain't there), I don't know why - thoughts?

birtony reviewed (changes requested):
Please, file another issue for cases like below before merging this PR @rjayroso 
![image](https://user-images.githubusercontent.com/33902374/111498144-14a11d00-8718-11eb-86ad-f3598475772e.png)

huynguyez reviewed (commented):
Nice for other blogs, but Blogspot image issue still there (32px).

rjayroso commented:
@huynguyez I can file an issue to fix the BlogSpot bug and merge this

I created another issue [here](https://github.com/Seneca-CDOT/telescope/issues/1975) of a potential fix for this

huynguyez reviewed (approved):
Since you gonna break it into another issue for Blogspot, this looks good to me!

chrispinkney reviewed (approved):
LGTM! Also very curious to see what happens with your blogspot idea 😄
